README.md

```diff
@@ -22,6 +22,8 @@ Javy is a beta project and will be under major development. We welcome feedback,
   install cmake`
 - Rosetta 2 if running MacOS on Apple Silicon, can be installed via
   `softwareupdate --install-rosetta`
+- Install and the `wasi-sdk` by running `./install-wasi-sdk.sh` or follow the
```

Comment:
`wasi-sdk` is definitely a system level dependency. I could've done a better job at making my point.

The configurability through the environment variable is fine, and we should keep it, this is important for a generic use  pulls the sys crates from crates.io for example. In a case like that, the installation script in its current form is not viable. 

My comment is geared towards _our_  own use case/implementation, mostly for developing, CI and publishing convenience. The intention of the top level Makefile is to orchestrate our build process,  so that testing building and publishing can be done with a single make command, hence my suggestion to use either of the mechanisms to install `wasi-sdk`, so that anyone that wants to get started with our implementation, can simply run `make` and have everything working for them instead of requiring them to run the installation script, same thing for the ci and publish workflows. Other use cases that would like to have a different CLI implementation, would have to pull in all the dependencies from crates.io and ensure that when building their CLI (i) wasi-sdk is set _somethere_ (ii) set `QUICKJS_WASM_SYS_WASI_SDK_PATH`. They get to decide how to orchestrate such workflow.

Reply:
> wasi-sdk is definitely a system level dependency.

👍

> The configurability through the environment variable is fine, and we should keep it, this is important for a generic use case that pulls the sys crate from crates.io for example.

👍 

> In a case like that, the installation script in its current form is not viable because it assumes the locality of the sys crate.

Technically true. I feel mildly uncomfortable about enabling that script to write to locations outside of the bounds of the repo (i.e., making system level changes), especially if we may be running it as part of a `make` target. We don't control or own those locations on the filesystem so I would argue we shouldn't be making assumptions about whether we can put a directory in there or not without consulting the end user. A package manager or the end user should own downloading things to locations outside the project directory since they would both know better about the state of the filesystem outside of the repository. I'm also okay with removing the installation scripts, it just adds a bit of toil.

> The intention of the top level Makefile is to orchestrate our build process, so that testing building and publishing can be done with a single make command, hence my suggestion to use either of the mechanisms to install wasi-sdk, so that anyone that wants to get started with our implementation, can simply run make and have everything working for them instead of requiring them to run the installation script, same thing for the ci and publish workflows

My understanding is that `make` shouldn't be responsible for installing system-level dependencies. E.g., we require installing `wasmtime-cli` and `cargo-wasi` outside of the `Makefile` (hypothetically `make` could handle installing those) and typically `make` invocations will exit with an error if an SDK is missing rather than trying to install it.